src/insertsort.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf -8 -*-
+
+from __future__ import unicode_literals
+from random import choice
+
+
+def insertion_sort(my_list):
+    """Simple sorting algorithm that builds a sorted list
+       one item at a time.
+    """
+    for i in range(1, len(my_list)):
+        current = my_list[i]
+        position = i
+        while((position > 0) and (my_list[position - 1] > current)):
+            my_list[position] = my_list[position - 1]
+            position = position - 1
+        if position != i:
```

Comment:
This check is an interesting thing to include. You are basically adding a comparison check at O(n) to avoid doing an assignment operation in the cases where your value-to-insert does not need to be inserted somewhere else. I'm not sure if this is more or less expensive.